adapters/repos/db/lsmkv/segment_serialization.go

```diff
@@ -163,7 +165,16 @@ func ParseReplaceNode(r io.Reader, secondaryIndexCount uint16) (segmentReplaceNo
 	return out, nil
 }
 
-func ParseReplaceNodeInto(r io.Reader, secondaryIndexCount uint16, out *segmentReplaceNode) error {
+func ParseReplaceNodeInto(r io.Reader, secondaryIndexCount uint16, out *segmentReplaceNode) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			debug.PrintStack()
+			err = errors.Wrapf(lsmkv.NotFound, "panic in ParseReplaceNodeInto: %v", r)
+		} else if err != nil {
```

Comment:
wouldn't wrap the err

Reply:
Agreed, I thought we had removed that again. E.g. the risk of breaking WAL recovery relying on an EOF, etc

Reply:
Yeah, this was just a bad cherry-pick conflict resolution. Removing